From b0b9c7831305818e3c46011540a7b145404f002d Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?Julian=20R=C3=BCth?= <julian.rueth@fsfe.org>
Date: Mon, 30 Aug 2021 01:00:15 +0200
Subject: [PATCH] Do not use -r flag unsupported on Windows

---
 Makefile.in      | 5 ++---
 Makefile.subdirs | 2 +-
 2 files changed, 3 insertions(+), 4 deletions(-)

diff --git a/Makefile.in b/Makefile.in
index 68169e93..efcaadf9 100644
--- a/Makefile.in
+++ b/Makefile.in
@@ -29,7 +29,6 @@ LIB_OBJS = $(patsubst %, build/%/*.o, $(BUILD_DIRS))
 
 LOBJS = $(patsubst %.c, build/%.lo, $(SOURCES))
 LIB_LOBJS = $(patsubst %, build/%/*.lo, $(BUILD_DIRS))
-MOD_LOBJS = $(patsubst %, build/%.lo, $(BUILD_DIRS))
 
 EXMP_SOURCES = $(wildcard examples/*.c)
 EXMPS = $(patsubst %.c, %, $(EXMP_SOURCES))
@@ -91,10 +90,10 @@ $(ANTIC_LIB): $(LOBJS) $(LIB_SOURCES) $(EXT_SOURCES) $(HEADERS) $(EXT_HEADERS) |
 	$(AT)$(foreach dir, $(BUILD_DIRS), mkdir -p build/$(dir); BUILD_DIR=../build/$(dir); export BUILD_DIR; MOD_DIR=$(dir); export MOD_DIR; $(MAKE) -f ../Makefile.subdirs -C $(dir) shared || exit $$?;)
 	$(AT)if [ "$(WANT_NTL)" -eq "1" ]; then \
 	  $(MAKE) build/interfaces/NTL-interface.lo; \
-	  $(CXX) $(ABI_FLAG) -shared $(EXTRA_SHARED_FLAGS) build/interfaces/NTL-interface.lo $(LOBJS) $(MOD_LOBJS) $(EXT_OBJS) -o $(ANTIC_LIB) $(LDFLAGS) $(LIBS2); \
+	  $(CXX) $(ABI_FLAG) -shared $(EXTRA_SHARED_FLAGS) build/interfaces/NTL-interface.lo $(LOBJS) $(LIB_LOBJS) $(EXT_OBJS) -o $(ANTIC_LIB) $(LDFLAGS) $(LIBS2); \
 	fi
 	$(AT)if [ "$(WANT_NTL)" -ne "1" ]; then \
-	  $(CC) $(ABI_FLAG) -shared $(EXTRA_SHARED_FLAGS) $(LOBJS) $(MOD_LOBJS) $(EXT_OBJS) -o $(ANTIC_LIB) $(LDFLAGS) $(LIBS2); \
+	  $(CC) $(ABI_FLAG) -shared $(EXTRA_SHARED_FLAGS) $(LOBJS) $(LIB_LOBJS) $(EXT_OBJS) -o $(ANTIC_LIB) $(LDFLAGS) $(LIBS2); \
 	fi
 	-$(AT)if [ "$(ANTIC_SOLIB)" -eq "1" ]; then \
 		$(LDCONFIG) -n "$(CURDIR)"; \
diff --git a/Makefile.subdirs b/Makefile.subdirs
index 53817d3f..41213d5b 100644
--- a/Makefile.subdirs
+++ b/Makefile.subdirs
@@ -32,7 +32,7 @@ TUNE = $(patsubst %.c, %$(EXEEXT), $(TUNE_SOURCES))
 
 all: shared static
 
-shared: $(MOD_LOBJ)
+shared: $(LOBJS)
 
 static: $(OBJS)
 
-- 
2.33.0

